services/src/kilda-core/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/FlowSegmentRepository.java

```diff
@@ -24,4 +25,6 @@
     Collection<FlowSegment> findByFlowIdAndCookie(String flowId, long flowCookie);
 
     long getUsedBandwidthBetweenEndpoints(SwitchId srcSwitchId, int srcPort, SwitchId dstSwitchId, int dstPort);
+
+    void lockSwitches(Switch... switches);
```

Comment:
The method operates with switches, so it goes to SwitchRepository.

Reply:
Fixed.